Std/Data/Array/Lemmas.lean

```diff
@@ -724,3 +724,105 @@ theorem all_eq_true_iff_forall_mem {l : Array α} : l.all p ↔ ∀ x, x ∈ l 
 /-! ### erase -/
 
 @[simp] proof_wanted erase_data [BEq α] {l : Array α} {a : α} : (l.erase a).data = l.data.erase a
+
+/-! ### swap -/
+
+@[simp] theorem get_swap_right (a : Array α) {i j : Fin a.size} :
+    (a.swap i j)[j.val]'(a.size_swap .. |>.symm ▸ j.isLt) = a[i] :=
+  by simp [swap, fin_cast_val]
+
+@[simp] theorem get_swap_left (a : Array α) {i j : Fin a.size} :
+    (a.swap i j)[i.val]'(a.size_swap .. |>.symm ▸  i.isLt) = a[j] :=
+  show _[i.val] = _ from
+    if he : ((Array.size_set _ _ _).symm ▸ j).val = i.val then
+      by simp [←he, fin_cast_val]
+    else
+      (Array.get_set_ne _ _ _ _ he).symm ▸ (show _[i.val] = _ by simp [get_set_ne])
+
+@[simp] theorem get_swap_of_ne (a : Array α) {i j : Fin a.size} (hp : p < a.size)
+    (hi : p ≠ i) (hj : p ≠ j) : (a.swap i j)[p]'(a.size_swap .. |>.symm ▸ hp) = a[p] :=
+  let vi := a.get i
+  let vj := a.get j
+
+  let j' := a.size_set i vj ▸ j
+  let a_ := a.set i vj
+  let a' := a_.set j' vi
```

Comment:
These are rather unhelpful names.

Reply:
I find term based proofs easier to read, but no problem,  will replace with tactic based proofs which wont have any auxiliary variables such as these.